core/database/foxx/api/user_router.js

```diff
@@ -516,9 +519,54 @@ router.get('/token/set', function(req, res) {
                         refresh: req.queryParams.refresh,
                         expiration: Math.floor(Date.now() / 1000) + req.queryParams.expires_in
                     };
-                    g_db._update(user_id, obj, {
-                        keepNull: false
-                    });
+                    if (token_type && other_token_data) {
```

Comment:
If one of this is defined and the other is not I would recommend throwing an error here because it indicates that the user doesn't realize the information they are providing will not be used unless both pieces of information are defined.